src/main/java/com/imsweb/mph/mpgroups/Mp2010HematopoieticGroup.java

```diff
@@ -321,7 +322,7 @@ public TempRuleResult apply(MphInput i1, MphInput i2) {
                         result.setPotentialResult(MphUtils.MpResult.MULTIPLE_PRIMARIES);
                         result.setMessageUnknownDiagnosisDate(this.getStep(), this.getGroupName());
                     }
-                    else if (MphConstants.DATE_VERIFY_APART == daysApart && latestDx > 0 && HematoUtils.isChronicToAcuteTransformation(MphConstants.COMPARE_DX_FIRST_LATEST == latestDx ? morph2 : morph1,
+                    else if (MphConstants.DATE_VERIFY_APART == daysApart && latestDx > 0 && HematoUtils.isChronicToAcuteTransformation(provider, MphConstants.COMPARE_DX_FIRST_LATEST == latestDx ? morph2 : morph1,
```

Comment:
If you feel strongly about keeping this solution, we can go with that.

But personally, I would have created a "RuleExecutionContext" class that takes an instance of "MphUtils" in its constructor (and caches it).  I would have moved the methods from HematoDbUtils to MphUtils, and I would do something like "context.getMphUtils().isChronicToAcuteTransformation()" in this method.

Or you could also add a "isChronicToAcuteTransformation()" convenience method to the context so you skip the "getMphUtils" call...

Personally I think this is a cleaner solution than having rules that call a static method. This fill all weirdly connected.